Design guidelines: Form and form control (#5047)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/components/form-control/form-control.md

## Usage
While form controls are useful in providing feedback on user input, they don't need to be implemented for every input instance. Only use form controls when it is essential to convey information to the user. For example, if any string is valid for a text input, or any selection in a list is acceptable, form controls are not necessary.

> Apply the `pf-m-form` modifier to all dropdowns and menu toggles used in a form. The `.pf-m-form` class modifies the menu toggle's border radius to match other form elements. Read more on form controls in the [Form guidelines](/components/forms/form/design-guidelines).

Reviewer: @nicolethoen @kmcfaul @mcoker this makes sense to me but do we need to mention props on the react components or is the CSS enough?

Reviewer 2: Personally I think it might be odd to only mention the class. "Apply the `pf-m-form` modifier, either by applying the class or using the isForm prop for React, to all menu toggles used in a form." Maybe? At the very least we should remove the "dropdown" from this as the modifier is only applicable to the menu toggle itself.

## Variations
While form control styles differ slightly depending on the component, they all use some combination of underlines, color, and icons to give users **feedback** on their inputs and selections. For example, if a user types an invalid string into a text box, the box changes state to indicate the error.

<div class="ws-docs-content-img">

> ![Types of form controls.](./img/form-control-variations.svg)

Reviewer 2: ```suggestion
![Types of form controls grouped by input, selection, and text area. Each form control is numbered in the image to correspond to a text list of descriptions.](./img/form-control-variations.svg)
```

Additionally, is there a reason why the image has 3 sections, each starting with numbering of 1, but only the first section is described after the image?

Author: Yes - Its showing the default untouched state of the component. Does that make sense? its not describing the name of the component itself, just that they are all existing in its default state @thatblindgeye

Use error controls when a user inputs incorrect information or selects an invalid option from a dropdown menu. Error indicators are often accompanied by helper text to further specify the user error. Read more on error messaging in the [forms guidelines](https://v4-archive.patternfly.org/v4/components/form/design-guidelines#errors-and-validation).

<div class="ws-docs-content-img">

> ![A gif showing an example of an input error.](./img/errorcontrols.gif)

Reviewer 2: ```suggestion
![An example if an input error. When a single letter "A" is entered into the "Name" field, an error helper text is rendered stating that the value must be at least 2 characters.](./img/errorcontrols.gif)
```
I'd argue that the helper text in the gif is something that should be mentioned immediately rather than as an input validation, but that isn't blocking right now.

</div>

#### Success controls
Use success controls to indicate valid user input for fields that require a specific form of input. In the example below, the name entered must be unique, so the success state indicates availability. Alternatively, an error control would be used to indicate a name that is already in use. **Only use success controls if the input must be validated.**

<div class="ws-docs-content-img">

> ![Example of a successful input.](./img/form-control-success.svg)

Reviewer 2: ```suggestion
![Example of a successful input, with a success variant helper text stating that the entered username is available.](./img/form-control-success.svg)
```
